config/samples/postgres.yaml

```diff
@@ -0,0 +1,12 @@
+apiVersion: addons.cluster.x-k8s.io/v1alpha1
+kind: HelmChartProxy
+metadata:
+  name: pg
+spec:
+  clusterSelector:
+    matchLabels:
+      postgresChart: enabled
+  version: 5.3.1
+  namespace: default
+  repoURL: oci://registry.developers.crunchydata.com/crunchydata
```

Comment:
```
// getHelmChartAndRepoName returns chartName, repoURL as per the format requirred in helm install/upgrade config.
// For OCI charts, chartName needs to have whole URL path. e.g. oci://repo-url/chart-name
```
Just to clarify based on this comment -- so we want to break up `oci://registry.developers.crunchydata.com/crunchydata/pgo` into two separate fields?

Reply:
That the best way to keep a single custom ressource format for both method
And based on helm oci doc this is not bad because there is a repo path and a chart name in the url

Reply:
User shall specify chart-path `oci://repo-url/chart-name` as `repoURL: oci://repo-url` and `chartName: chart-name` in HCP CR. This format is consistent with other types of charts as well (e.g. `https://repo-url/chart-name` as `repoURL: https://repo-url` and `chartName: chart-name`). This is in-line with other addons controller like fluxcd, argo (please [check](https://fluxcd.io/flux/guides/helmreleases/#helm-oci-repository)) so users won't require any learning path if switched from other tools to CAAPH.

What's the best place to add this documentation? [this](https://github.com/kubernetes-sigs/cluster-api-addon-provider-helm/blob/main/docs/quick-start.md)?

Reply:
Sounds good. And I think that's a reasonable place to document this change. We could potentially add a note in the CRD field description comment about it too if you think that wouldn't make it too long.

Reply:
sure, done. Please check.